packages/api/amplify_api/lib/src/graphql/ws/web_socket_connection.dart

```diff
@@ -92,6 +152,7 @@ class WebSocketConnection implements Closeable {
       protocols: webSocketProtocols,
     );
     _subscription = getStreamSubscription(_channel!.stream);
+    _network = getStreamNetwork();
   }
 
   void _connectionError(ApiException exception) {
```

Comment:
I think we need to check if `_connectionReady` is completed here before calling `completeError`. For ex, line 104 calls `completeError` and then immediately calls this line which will result in a `StateError` being thrown.